core/src/layers/4-lsm/tx_lsm_tree.rs

```diff
@@ -1,937 +1,958 @@
-//! Transactional LSM-tree.
+//! Transactional LSM-Tree.
 //!
-//! API: format(), recover(), get(), put(), get_range(), put_range(), commit(), discard()
+//! API: `format()`, `recover()`, `get()`, `put()`, `get_range()`, `sync()`
 //!
-//! Responisble for managing two `MemTable`s, a `TxLogStore` to
-//! manage tx logs (WAL and SSTs). Operations are executed based
+//! Responsible for managing two `MemTable`s, WAL and SSTs as `TxLog`s
+//! backed by a `TxLogStore`. All operations are executed based
 //! on internal transactions.
-use crate::layers::bio::{BlockBuf, BlockId};
-use crate::layers::log::{TxLog, TxLogId, TxLogStore};
+use super::compaction::Compactor;
+use super::mem_table::{MemTableManager, ValueEx};
+use super::range_query_ctx::RangeQueryCtx;
+use super::sstable::SSTable;
+use super::wal::{WalAppendTx, BUCKET_WAL};
+use crate::layers::bio::BlockSet;
+use crate::layers::log::{TxLogId, TxLogStore};
+use crate::os::{BTreeMap, RwLock};
+use crate::prelude::*;
 use crate::tx::Tx;
 
-use serde::{Deserialize, Serialize};
-use spin::{Mutex, RwLock};
-use std::collections::BTreeMap;
-use std::fmt::Debug;
-use std::io::Read;
-use std::marker::PhantomData;
-use std::mem::size_of;
-use std::ops::{Range, RangeBounds};
-use std::sync::atomic::{AtomicU64, AtomicU8, Ordering};
-use std::sync::Arc;
-
-const BUCKET_WAL: &str = "WAL";
-const BUCKET_L0: &str = "L0";
-const BUCKET_L1: &str = "L1";
-const BLOCK_SIZE: usize = 0x1000;
-
-static MASTER_COMMIT_ID: AtomicU64 = AtomicU64::new(0);
-
-/// A LSM-tree built upon `TxLogStore`.
-pub struct TxLsmTree<K, V, D> {
-    mem_tables: [Arc<RwLock<MemTable<K, V>>>; 2],
-    immut_idx: AtomicU8,
-    tx_log_store: Arc<TxLogStore<D>>,
-    append_tx: AppendTx<K, V, D>,
+use core::hash::Hash;
+use core::ops::{Add, RangeInclusive, Sub};
+use core::sync::atomic::{AtomicU64, Ordering};
+use pod::Pod;
+
+// TODO: Use `Thread` in os module
+#[cfg(feature = "occlum")]
+use sgx_tstd::thread;
+#[cfg(feature = "std")]
+use std::thread;
+
+// XXX: Master sync ID should be stored in external trusted storage
+pub static MASTER_SYNC_ID: AtomicU64 = AtomicU64::new(0);
```

Comment:
I am confused. First, the main memory should keep a copy of master sync ID for fast access. The master sync ID should only be fetched from the trusted store when recovering a `TxLsmTree` from the disk. So I don't understand the point of the comment. Second, this ID should be unique for each instance of `TxLsmTree`. I don't understand why this is kept as a global value.